### ISSUE TITLE
How do I make new fonts for this library?

### ISSUE THREAD

**Issue by rogerclarkmelbourne** [bitbank2/uc1701#2]
How did you make these fonts ?

I would like to add some different fonts, but don’t know what program creates this form of data.

Are you using the EA Font editor?
If so is there any way to get the old non commercial version? As the new one seems to cost $$$$$

**Comment by bitbank2** [comment]
Fonts are a tricky topic because of licensing issues. I got some font data from an IBM monochrome display adapter and rotated the bits to the orientation of the display (using my own code). I don't have any recommendations for you because I'm not a font expert. The format I use is the same as the display memory: bytes are oriented vertically with the LSB as the top pixel.

**Comment by rogerclarkmelbourne** [comment]
Thanks.

I eventually worked out the format, but it took a while, because the "pages" go down vertically and the stripes inside the page go from left to right, but the bit order is the LSBit at the bottom, which confused me for a while.

I found some fonts in a library on MBED, which seem to use the same format

https://os.mbed.com/users/Anaesthetix/code/UC1701/file/7494bdca926b/

Which makes me think that there must be a font generator somewhere which makes them in that format.

Someone recommend that I use an old Windows program called RaysFontEditor, but its output format is totally different from the data that either your library or the MBED library use.

The data format in yours, and the MBED lib, is optimised for this display to minimise processing.

I had initially thought I would write a modify library to read the data created by RaysFontEditor, but I realised its a big overhead when rendering the data to have to convert it each time.

So I've ended up writing a small utility in C# that converts 8 x 16 fonts from RaysFontEditor to the format that you library uses e.g. does the bit rotation in the byes to vertical raster in the appropriate groups etc.
But my converter only works for that size, and I'll need to modify it some time to handle other sizes.


I've also ended up re-writing the text drawing part of your library, so that I can position the text at any Y location, not just multiples of 8.

I'm using an offscreen buffer, to OR in the data, because the display module I'm using doesn't expose the Read / Write line :-(

So my version always writes to the off-screen buffer and I've added an Update() function which needs to be called to write the buffer to the screen.

This seems to work relatively well.

When my changes are a bit cleaner I'll post them here, in case you want to use them, but it is a bit of a radical change and the things which I need to position the text of the different sized fonts as close to each other as possible to make the best use of the display..

Anyway, thanks again for fixing the big font issue. I've not had time to download and test it yet, but I hope to download it in this evening.
